compiler/rustc_target/src/asm/powerpc.rs

```diff
@@ -113,6 +115,40 @@ def_regs! {
         f29: freg = ["f29", "fr29"],
         f30: freg = ["f30", "fr30"],
         f31: freg = ["f31", "fr31"],
+        v0: vreg = ["v0"],
+        v1: vreg = ["v1"],
+        v2: vreg = ["v2"],
+        v3: vreg = ["v3"],
+        v4: vreg = ["v4"],
+        v5: vreg = ["v5"],
+        v6: vreg = ["v6"],
+        v7: vreg = ["v7"],
+        v8: vreg = ["v8"],
+        v9: vreg = ["v9"],
+        v10: vreg = ["v10"],
+        v11: vreg = ["v11"],
+        v12: vreg = ["v12"],
+        v13: vreg = ["v13"],
+        v14: vreg = ["v14"],
+        v15: vreg = ["v15"],
+        v16: vreg = ["v16"],
+        v17: vreg = ["v17"],
+        v18: vreg = ["v18"],
+        v19: vreg = ["v19"],
+        // FIXME: In AIX, v20-v31 are reserved or nonvolatile depending on the mode.
+        // https://www.ibm.com/docs/en/aix/7.3?topic=concepts-aix-vector-programming
```

Comment:
@ecnelises @bzEq: Is the default in Rust the default Vector enabled mode, as the name implies? Also, is there a way provided for the compiler to understand the current mode?

(If the first is yes and the second is no, it would be sufficient to simply reject the use of v20-v31 as reserved. If the first is no, this code is fine as is.)

Reply:
It should refer to `vec-extabi`, which is the default. See https://github.com/rust-lang/rust/blob/8f8bee4f60d9d3769f75c70d558c27a95761c554/compiler/rustc_target/src/spec/base/aix.rs#L6.